include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -245,10 +188,44 @@ class alignment_algorithm :
 
         std::vector<alignment_result<result_t>> results{};
         for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
-            results.emplace_back((*this)(idx, get<0>(sequence_pair), get<1>(sequence_pair)));
+            results.emplace_back(compute_single_pair(idx, get<0>(sequence_pair), get<1>(sequence_pair)));
 
         return results;
     }
+    //!\}
+private:
+
+    /*!\brief Computes the pairwise sequence alignment for a single pair of sequences.
+     * \tparam sequence1_t The type of the first sequence; must model std::ranges::forward_range.
+     * \tparam sequence2_t The type of the second sequence; must model std::ranges::forward_range.
+     *
+     * \param[in] idx The index of the current processed sequence pair.
+     * \param[in] sequence1 The first sequence (or packed sequences).
+     * \param[in] sequence2 The second sequence (or packed sequences).
+     *
+     * \returns A seqan3::alignment_result with the requested alignment outcomes.
+     *
+     * \throws std::bad_alloc during allocation of the alignment matrices or
+     *         seqan3::invalid_alignment_configuration if an invalid configuration for the given sequences is detected.
+     *
+     * \details
+     *
+     * Uses the standard dynamic programming algorithm to compute the pairwise sequence alignment. The space and
+     * runtime complexities depend on the selected configurations (see below).
```

Comment:
You refer to the removed the description of complexity.

Reply:
ok I removed it because it does not add something to the understanding of the function and the public interface already documents this.